### DISCUSSION TITLE
Using cucumber format to produce an HTML report with Cucumber HTML reporter

### DISCUSSION THREAD

**Discussion by robert-ovens** [cucumber/godog#557] [Q&A]
Hi all,

Versions
* go version go1.20.3 linux/amd64
* github.com/cucumber/godog v0.12.6

I am attempting to produce an HTML report of the godog output. The output does contain the desired json, however it also outputs the default output appended to the end.

A sample test will include the function

```go
func TestFeatures(t *testing.T) {

	suite := godog.TestSuite{
		TestSuiteInitializer: InitializeTestSuite,
		ScenarioInitializer:  InitializeScenario,
		Options: &godog.Options{
			Format:   "cucumber",
			Paths:    []string{"../features"},
			TestingT: t, // Testing instance that will run subtests.
		},
	}

	if suite.Run() != 0 {
		t.Fatal("non-zero status returned, failed to run feature tests")
	}
}
```

And the test is executed thusly `go test ./api`. `./api/` is the folder containing the tests. The output is;

```json
[
    {
        "uri": "../features/documents.feature",
        "id": "document-generation",
        ...<redacted the output>
    }
]
--- FAIL: TestFeatures (0.00s)
    --- FAIL: TestFeatures/The_user_provides_a_valid_PD_and_asks_for_it_as_HTML (0.00s)
        suite.go:451: 
    ...<redacted the output>
FAIL
```
So, the desired json for conversion with Cucumber HTML Reporter, but additional output which I do not want. My goal is to generate the test results as cucumber json so I can gnerate HTML reports in our CI/CD pipeline.

* I may not be using `test` correctly (new to Go)
* I may not be using godog correctly

Thanks in advance

**Comment by robert-ovens** [comment]
I think I understand now how to proceed. It really involved looking through the godog source code :-)

The `TestFeatures` has been modified slightly;

```go
func TestFeatures(t *testing.T) {

	f, err := os.Create("./report.json")
	check(err)

	suite := godog.TestSuite{
		TestSuiteInitializer: InitializeTestSuite,
		ScenarioInitializer:  InitializeScenario,
		Options: &godog.Options{
			Format:   "cucumber",
			Output:   f,
			Paths:    []string{"features"},
			TestingT: t, // Testing instance that will run subtests.
		},
	}

	if suite.Run() != 0 {
		t.Fatal("non-zero status returned, failed to run feature tests")
	}
	f.Close()
}
```

- Created a file handle which is passed to the test suite as an options flag
- Moved the features folder to the package under test as the cucumber formatter does not like relativized paths

Props to godog authors - a very nice library :-)

**Reply by yogiis** [reply]
You can also use this for a simpler `Format:   "pretty,cucumber:./report/cucumber.json",`
